File: SwapNodesInPairs.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        if(head == nullptr || head->next == nullptr) {
            return head;
        }
        ListNode* tmp = head->next;
        head->next = tmp->next;
        tmp->next = head;
        head = tmp;
        ListNode *headTmp = swapPairs(head->next->next);
        if(head->next->next != nullptr && head->next != nullptr) {
            head->next->next = headTmp;
        }
        return head;
        
    }
};

File: SwapNodesInPairs.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode SwapPairs(ListNode head) {
        if(head == null || head.next == null) {
            return head;
        }
        var tmp = head.next;
        head.next = tmp.next;
        tmp.next = head;
        head = tmp;
        ListNode newHead = SwapPairs(head.next.next);
        if(head.next != null && head.next.next != null) {
            head.next.next = newHead;
        }
        return head;
    }
}
